pandas/core/reshape/merge.py

```diff
@@ -1477,6 +1475,19 @@ def get_join_indexers(
         right_keys
     ), "left_key and right_keys must be the same length"
 
+    left_n = len(left_keys[0])
```

Comment:
can you add a comment on what you are checking here (e.g. fast-path for empty left/right)

Reply:
comment added